mempool/reactor.go

```diff
@@ -276,7 +253,7 @@ func (memR *Reactor) broadcastTxRoutine(peer p2p.Peer) {
 		// NOTE: Transaction batching was disabled due to
 		// https://github.com/tendermint/tendermint/issues/5796
 
-		if !memR.isSender(memTx.tx.Key(), peer.ID()) {
+		if !memTx.isSender(peer.ID()) {
```

Comment:
This is the main reason of this PR. When reading the sender, it must be available immediately to not send back the tx. By storing the sender in the reactor, a small delay was introduced by the callback.